### PR TITLE
Add compilers.rules

### PR DESCRIPTION
**Pull request by JohnRTitor**
C and C++, Go, Java, and Rust compilers are added
Also added Bazel, Cmake, Ninja, and Meson build systems

### CODE REVIEW

**Comment by ptr1337** [comment]
> C and C++, Go, Java, and Rust compilers are added Also added Bazel, Cmake, Ninja, and Meson build systems

We have explicity removed compiler, because when it spawns many tasks/threads, ananicy-cpp will quite much cpu power at some point.
Additionally there was a regression, in the BORE compiler, when the compiler rules were set. We maybe need to retest this.

**Comment by aarrayy** [comment]
On my testing, it got completely fixed on BORE +4.5.2 with -9, -1, 1, 5, 10, and 19 niceness values with the scan interval set to 1 second.

**Comment by JohnRTitor** [comment]
It should be stable on nice 10 and 19.

**Comment by ptr1337** [comment]
Maybe linkers, like lld, ld, ld.bfd, mold can be also added.

**Comment by ptr1337** [comment]
We should also check, if this brings a noticeable performance regression.

**Comment by JohnRTitor** [comment]
> Maybe linkers, like lld, ld, ld.bfd, mold can be also added.

Added. Seems fine on my end, then again, it should be tested by the reviewers.

File: 00-default/compilers.rules
Comment (ptr1337) [inline 00-default/compilers.rules:33]: Please add newline here

File: 00-types.types
Comment (ptr1337) [inline 00-types.types:32]: Should we maybe put it a bit higher like 11 to 13?
@aarrayy WDYT?

**Review by ptr1337 — APPROVED** [review]

File: 00-default/compilers.rules
Comment (JohnRTitor) [reply 00-default/compilers.rules]: Done.

**Comment by ptr1337** [comment]
@aarrayy
Could you please make some testing with 13, 11 and  9?

**Comment by aarrayy** [comment]
> @aarrayy Could you please make some testing with 13, 11 and 9?

What kind of testing? How much time does it adds? or...

**Comment by ptr1337** [comment]
> > @aarrayy Could you please make some testing with 13, 11 and 9?
> 
> What kind of testing? How much time does it adds? or...

Likely if BORE shows issue together with it again.
Ill also start testing it on my machine for some days.

**Comment by aarrayy** [comment]
> > > @aarrayy Could you please make some testing with 13, 11 and 9?
> > 
> > 
> > What kind of testing? How much time does it adds? or...
> 
> Likely if BORE shows issue together with it again. Ill also start testing it on my machine for some days.

Sure, i'll do some testing with 13 and 9

**Comment by JohnRTitor** [comment]
> I currently use 10 and its fine on my system, it leaves room for multitasking without hurting too much the build process itself, for example web browsing while compiling the linux kernel with thin lto i got a total time of 22 mins

I agree. Though I am using 9 on normal cachyos kernel with sched-ext rustland

**Comment by ptr1337** [comment]
Ill also need to do in the coming days some throughput tests, since ananicy-cpp uses quite much cpu time, when applying this compiler rules

**Comment by aarrayy** [comment]
BORE is totally fine with any nice value, on 4.5.2 i tested from 1, 3, 6, 9, 12, 19 and couldn't manage to get a lockup and ananicy is set to an scan interval of 1 second so it has to happen no matter what, 5.0.3 is the same history

Passed on 10 and 13

**Comment by vnepogodin** [comment]
> ananicy is set to an scan interval of 1 second

there is no scan interval. it does push into buffer on every sched_exec and sched_fork.

the poll from that buffer is done "non-stop" with non-configurable timeout of 500ms

**Comment by aarrayy** [comment]
> > ananicy is set to an scan interval of 1 second
> 
> there is no scan interval. it does push into buffer on every sched_process_exec and sched_process_fork.
> 
> the poll from that buffer is done "non-stop" with non-configurable timeout of 500ms

I was referring to check_freq in ananicy.conf that by default is set to 15 seconds and i personally use 1 second

**Comment by vnepogodin** [comment]
> check_freq

thats only for when flag `--manual-scanning` is being passed

**Comment by aarrayy** [comment]
> > check_freq
> 
> thats only for when flag `--manual-scanning` is being passed

Oh.

**Comment by JohnRTitor** [comment]
So... should I modify the config "nice" "latency-nice" to 13 in this PR?

File: 00-types.types
Comment (JohnRTitor) [reply 00-types.types:32]: I have been daily driving this at 13 for two weeks now, it is stable. 13 should leave some more headroom.

**Comment by JohnRTitor** [comment]
@aarrayy @ptr1337 I have been daily driving at 13 nice for two weeks now. It stays stable on high load. 

Waiting for your final approval :)

**Comment by aarrayy** [comment]
> @aarrayy @ptr1337 I have been daily driving at 13 nice for two weeks now. It stays stable on high load.
> 
> Waiting for your final approval :)

I approve these rules, i haven't had any lockup with any nice value since BORE +4.5.2

**Pull request merged**
